lib/src/main/kotlin/io/provenance/core/KeyEntityManager.kt

```diff
@@ -1,18 +1,17 @@
 package io.provenance.core
 
 import io.provenance.entity.KeyEntity
-import java.util.UUID
 
 class KeyEntityManager {
     private val entities: MutableMap<String, KeyEntity> = mutableMapOf()
-    private val plugins: MutableSet<Plugin> = mutableSetOf()
+    private val plugins: MutableSet<Plugin<*>> = mutableSetOf()
 
-    fun register(plugin: Plugin) =
+    fun register(plugin: Plugin<*>) =
         plugins.add(plugin)
 
-    fun get(entity: String, pluginSpec: Any): KeyEntity {
+    fun <T: PluginConfig> get(entity: String, config: T): KeyEntity {
         if (entity !in entities) {
-            entities[entity] = plugins.firstOrNull { it.supports(pluginSpec) }?.fetch(pluginSpec)
+            entities[entity] = plugins.filterIsInstance<Plugin<T>>().singleOrNull()?.fetch(entity, config)
                 ?: throw IllegalStateException("$entity has no supported plugins.")
```

Comment:
Nit: suggest changing error message to be more accurate due to introduction of `singleOrNull`, e.g.
```suggestion
            entities[entity] = plugins.filterIsInstance<Plugin<T>>().singleOrNull()?.fetch(entity, config)
                ?: throw IllegalStateException("$entity did not have exactly one supported plugin.")
```
Either that or maybe adding a check in `register` or changing `plugins` to a map?